x-pack/plugins/infra/public/components/asset_details/tabs/overview/metrics/metric_chart.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useMemo } from 'react';
+import { Action } from '@kbn/ui-actions-plugin/public';
+import {
+  EuiIcon,
+  EuiPanel,
+  EuiI18n,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  useEuiTheme,
+} from '@elastic/eui';
+import { css } from '@emotion/react';
+import { TypedLensByValueInput } from '@kbn/lens-plugin/public';
+import type { DataView } from '@kbn/data-views-plugin/public';
+import { TimeRange } from '@kbn/es-query';
+import {
+  buildCombinedHostsFilter,
+  buildExistsHostsFilter,
+} from '../../../../../utils/filters/build';
+import { LensWrapper } from '../../../../../common/visualizations/lens/lens_wrapper';
+import { useLensAttributes, type Layer } from '../../../../../hooks/use_lens_attributes';
+import type { FormulaConfig, XYLayerOptions } from '../../../../../common/visualizations';
+
+export interface MetricChartProps extends Pick<TypedLensByValueInput, 'id' | 'overrides'> {
+  title: string;
+  layers: Array<Layer<XYLayerOptions, FormulaConfig[]>>;
+  dataView?: DataView;
+  timeRange: TimeRange;
+  nodeName: string;
+}
+
+const MIN_HEIGHT = 250;
+
+export const MetricChart = ({
+  id,
+  title,
+  layers,
+  nodeName,
+  timeRange,
+  dataView,
+  overrides,
+}: MetricChartProps) => {
+  const { euiTheme } = useEuiTheme();
+
+  const { attributes, getExtraActions, error } = useLensAttributes({
+    dataView,
+    layers,
+    title,
+    visualizationType: 'lnsXY',
+  });
+
+  const filters = useMemo(() => {
+    return [
+      buildCombinedHostsFilter({
+        field: 'host.name',
+        values: [nodeName],
+        dataView,
+      }),
+      buildExistsHostsFilter({ field: 'host.name', dataView }),
+    ];
+  }, [dataView, nodeName]);
+
+  const extraActions: Action[] = useMemo(
+    () =>
+      getExtraActions({
+        timeRange,
+        filters,
+      }),
+    [timeRange, filters, getExtraActions]
+  );
+
+  const loading = !attributes;
+
+  return (
+    <EuiPanel
+      borderRadius="m"
+      hasShadow={false}
+      hasBorder
+      paddingSize={error ? 'm' : 'none'}
+      css={css`
+        min-height: calc(${MIN_HEIGHT}px + ${euiTheme.size.l});
+        position: relative;
+      `}
+      data-test-subj={`assetDetailsMetricsChart${id}`}
+    >
+      {error ? (
+        <EuiFlexGroup
+          style={{ minHeight: '100%', alignContent: 'center' }}
+          gutterSize="xs"
+          justifyContent="center"
+          alignItems="center"
+          direction="column"
+        >
+          <EuiFlexItem grow={false}>
+            <EuiIcon type="warning" />
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiText size="s" textAlign="center">
+              <EuiI18n
```

Comment:
Nice, new EUI component (at least for me 😅) 🙂

Reply:
Kind of copied this from the other `metric_chart` component. I will change this to use `FormattedMessage` to keep consistency.